src/test/java/org/openhab/tools/analysis/checkstyle/test/KarafFeatureCheckTest.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle.test;
+
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.POM_XML_FILE_NAME;
+
+import java.io.File;
+import java.text.MessageFormat;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openhab.tools.analysis.checkstyle.KarafFeatureCheck;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheckTest;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.api.Configuration;
+
+/**
+ * Tests for {@link KarafFeatureCheck}
+ *
+ * @author Svilen Valkanov - Initial contribution
+ */
+public class KarafFeatureCheckTest extends AbstractStaticCheckTest {
+
+    private static final String MSG_MISSING_BUNDLE_IN_FEATURE_XML = "Bundle with ID '{0}' must be added in {1}";
+    private static final String TEST_DIRECTORY = "karafFeatureCheck";
+    private static final DefaultConfiguration CONFIGURATION = createCheckConfig(KarafFeatureCheck.class);
+
+    @Override
+    protected DefaultConfiguration createCheckerConfig(Configuration config) {
+        DefaultConfiguration configParent = new DefaultConfiguration("root");
+        configParent.addChild(config);
+        return configParent;
+    }
+
+    @BeforeClass
+    public static void setUp() {
+        CONFIGURATION.addAttribute("featureXmlPath", "feature/feature.xml");
+    }
+
+    @Test
+    public void testIncludedBundle() throws Exception {
+        verify("includedBundle", ArrayUtils.EMPTY_STRING_ARRAY);
+    }
+
+    @Test
+    public void testIncludedBundleWithParentGroupIdOnly() throws Exception {
+        verify("includedBundleWithParentGroupIdOnly", ArrayUtils.EMPTY_STRING_ARRAY);
+    }
+
+    @Test
+    public void testInvalidBundle() throws Exception {
+        verify("invalidBundle", ArrayUtils.EMPTY_STRING_ARRAY);
+        // Error will be logged on the console
```

Comment:
There are actually ways to validate this by adding a log appender for the specific class and capturing the specific message.